python/cuml/common/base.pyx

```diff
@@ -19,68 +19,123 @@
 # cython: embedsignature = True
 # cython: language_level = 3
 
-
+import cuml
 import cuml.common.handle
 import cuml.common.cuda
 import inspect
 
+from cudf.core import Series, DataFrame
+from cuml.common.array import Array
+from cupy import ndarray as cupyarray
+from numba.cuda import is_cuda_array
+from numpy import ndarray as numpyarray
+
+_input_type_to_str = {
+    numpyarray: 'numpy',
+    cupyarray: 'cupy',
+    Series: 'cudf',
+    DataFrame: 'cudf'
+}
+
 
 class Base:
     """
     Base class for all the ML algos. It handles some of the common operations
     across all algos. Every ML algo class exposed at cython level must inherit
     from this class.
 
+    Typical estimator design using Base requires three main things:
+
+    1. Call the base init method explicitly from inheriting estimators in their
```

Comment:
"call the base __init__" method just to be explicit

Reply:
don't you love how github/markdown make `__init__` just bold without the underscores sometimes, obscuring the meaning of a comment? :P